src/features/Posts/postSlice.js

```diff
@@ -44,6 +44,46 @@ export const createPost = createAsyncThunk(
   }
 );
 
+export const editPost = createAsyncThunk(
+  "posts/edit",
+  async ({ postId, editedContent }, { rejectWithValue }) => {
+    try {
+      const {
+        data: { response },
+      } = await axios({
+        method: "POST",
+        url: `${API_URL}/posts/${postId}/editpost`,
+        headers: {
+          authorization: "Bearer " + JSON.parse(localStorage.getItem("token")),
+        },
```

Comment:
Working fine as per the functionality. Just one suggestion you can set the token as a default header once user logged in and then remove on logout , instead of writing it for different requests which requires token. you can read about it here https://axios-http.com/docs/config_defaults